az-core/src/main/java/azkaban/Constants.java

```diff
@@ -153,6 +153,9 @@
   // Overridable plugin load properties
   public static final String AZ_PLUGIN_LOAD_OVERRIDE_PROPS = "azkaban.plugin.load.override.props";
 
+  // File containing param override configs
```

Comment:
It will be helpful to specify the intended priority as well for the properties within this file.

Reply:
I dont understand. It is pretty evident from the name and comment.

Reply:
1.
For a directory structure, property files in Proj_Dir used to have lower precedence than A. For the newly introduced file, this will no longer be true

Proj_Dir
  basic.properties
  param_override.properties
  A/
    foo_a.properties
    foo.job
    
i.e. 
(a). param_override.properties precedence will be higher than foo_a.properties. 
(b). foo_a.properties precedence will be higher than that of basic.properties.

Hence this should be very well documented.


2. 
Furthermore, so far UI specified parameters take the highest precedence. I believe for this file, this won't be the case - which can be very counter intuitive for a user making changes in the UI.

I would say the UI should always have higher precedence. Any reason why that would be undesired in this case? Assuming, it's a valid reason, that reason also ought to be documented.

Reply:
Resolving as we discussed it offline

Reply:
In addition to add this to the documentation (https://azkaban.github.io/azkaban/docs/latest/), a comment over here will be very helpful when someone else reads this code. It will be good to add intent as well as the rationale as to why we wanted a file based property to override a UI-specified parameter.

Reply:
I will raise a followup PR with comments.